model/headers/headers.go

```diff
@@ -213,8 +213,8 @@ func (h Header) MarshalBinary() (data []byte, err error) {
 	if err != nil {
 		return nil, err
 	}
-	for _, headerPacket := range h.HeaderPackets {
-		marshalledHeaderPacket, err := headerPacket.MarshalBinary()
+	for i := range h.HeaderPackets {
```

Comment:
Not wrong, but this change does not seem needed if no longer keeping the nonces in the header.

Reply:
It is needed because otherwise the new Nonce is not reachable in other function.